### PR TITLE
Improve validation against negative keys

### PR DESCRIPTION
**Pull request by infiloop2**
Only allow non-negative numbers in observations

### CODE REVIEW

**Review by ferglor — DISMISSED** [review]

**Comment by ferglor** [comment]
Looks good, are there any test cases we could add to these new validations?

File: internal/keepers/util_test.go [hunk 1]
```diff
@@ -180,6 +180,20 @@ func Test_observationsToUpkeepKeysError(t *testing.T) {
...
+			BlockKey: chain.BlockKey("-1"), // Should be non negative
+			UpkeepIdentifiers: []ktypes.UpkeepIdentifier{
+				ktypes.UpkeepIdentifier("2"),
+				ktypes.UpkeepIdentifier("3"),
+			},
+		},
+		{
+			BlockKey: chain.BlockKey("1"), // Should be non negative
+			UpkeepIdentifiers: []ktypes.UpkeepIdentifier{
+				ktypes.UpkeepIdentifier("-2"),
```
Comment (ferglor) [inline internal/keepers/util_test.go:193]: Should this line should be commented to state that the UpkeepIdentifier should be non-negative?

**Review by ferglor — APPROVED** [review]
🎉

File: pkg/chain/upkeep_observation.go [hunk 2]
```diff
@@ -53,6 +56,9 @@ func (u *UpkeepObservation) validate(uo upkeepObservation) error {
 		if uiInt.String() != string(ui) {
 			return ErrInvalidUpkeepIdentifier
 		}
+		if uiInt.Cmp(big.NewInt(0)) == -1 {
```
Comment (EasterTheBunny) [inline pkg/chain/upkeep_observation.go:59]: Should this be `<= 0` instead since zero value identifiers are also invalid?

Comment (infiloop2) [reply pkg/chain/upkeep_observation.go:59]: I considered it, technically upkeep ID is uint256 so it can be 0. I'll add it for the blockKey though

**Review by ferglor — APPROVED** [review]

File: pkg/chain/upkeep_observation.go (hunk 2 above)
Comment (EasterTheBunny) [reply pkg/chain/upkeep_observation.go:60]: Ahh. ok. Sounds good.

**Pull request merged**